From 1532ac03481b6750556eb76aca79c569ab553152 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Wed, 18 Jun 2014 21:02:15 +1000
Subject: [PATCH 03/10] =?UTF-8?q?Abort=20early=20when=20trying=20to=20open=20?=
 =?UTF-8?q?a=200=20bytes=20file=20size=20if=20it=E2=80=99s=20not=20being=20w?=
 =?UTF-8?q?ritten=20to?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This could cause a 10s delay liveTV startup as backend attempts to open the first dummy recording
---
 mythtv/libs/libmythtv/fileringbuffer.cpp |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/mythtv/libs/libmythtv/fileringbuffer.cpp b/mythtv/libs/libmythtv/fileringbuffer.cpp
index d3e2927..e875bfb 100644
--- a/mythtv/libs/libmythtv/fileringbuffer.cpp
+++ b/mythtv/libs/libmythtv/fileringbuffer.cpp
@@ -242,6 +242,13 @@ bool FileRingBuffer::OpenFile(const QString &lfilename, uint retry_ms)
                     lasterror = 2;
                     close(fd2);
                     fd2 = -1;
+                    if (ret == 0 && openAttempts > 5 &&
+                        !gCoreContext->IsRegisteredFileForWrite(filename))
+                    {
+                        // file won't grow, abort early
+                        break;
+                    }
+
                     if (oldfile)
                         break; // if it's an old file it won't grow..
                     usleep(10 * 1000);
-- 
1.7.10.2

